vsix/CppWinRTPackageTemplateWizard.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Evaluation;
+using Microsoft.VisualStudio.TemplateWizard;
+using System.Windows.Forms;
+
+using DTEProject = EnvDTE.Project;
+using DTEProjectItem = EnvDTE.ProjectItem;
+
+namespace Microsoft.Windows.CppWinRT
+{
+    internal sealed class CppWinRTPackageTemplateWizard : IWizard
+    {
+        private IWizard wizardImpl = null;
+
+        private void ShowExceptionDialog(Exception ex)
+        {
+            string text = $"{ex.GetType().FullName}: {ex.Message}\r\n\r\n{ex.StackTrace}";
+            MessageBox.Show(text, "Caught Exception", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            throw new WizardBackoutException();
+        }
+
+        public void BeforeOpeningFile(DTEProjectItem projectItem)
+        {
+            try
+            {
+                wizardImpl?.BeforeOpeningFile(projectItem);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectFinishedGenerating(DTEProject project)
+        {
+            try
+            {
+                VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Is it correct to throw an exception in a try-catch block in a VS extension?

Reply:
See my commit message on 4c0efeb.